nova_vm/src/heap/heap_bits.rs

```diff
@@ -371,33 +311,96 @@ impl WorkQueues {
     }
 
     pub fn is_empty(&self) -> bool {
-        self.modules.is_empty()
-            && self.scripts.is_empty()
-            && self.realms.is_empty()
-            && self.declarative_environments.is_empty()
-            && self.function_environments.is_empty()
-            && self.object_environments.is_empty()
-            && self.e_2_4.is_empty()
-            && self.e_2_6.is_empty()
-            && self.e_2_8.is_empty()
-            && self.e_2_10.is_empty()
-            && self.e_2_12.is_empty()
-            && self.e_2_16.is_empty()
-            && self.e_2_24.is_empty()
-            && self.e_2_32.is_empty()
-            && self.arrays.is_empty()
-            && self.array_buffers.is_empty()
-            && self.bigints.is_empty()
-            && self.errors.is_empty()
-            && self.dates.is_empty()
-            && self.bound_functions.is_empty()
-            && self.builtin_functions.is_empty()
-            && self.ecmascript_functions.is_empty()
-            && self.numbers.is_empty()
-            && self.objects.is_empty()
-            && self.regexps.is_empty()
-            && self.strings.is_empty()
-            && self.symbols.is_empty()
+        let Self {
```

Comment:
**note:** This was an unreliable function as it was not being updated when more stuff was being added to WorkQueues.

Destructuring the borrow fixes that problem.